Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -43,23 +44,67 @@ public void Razor_ForProductProject_CheckErrorLogProperties()
             // Arrange
             var rootInputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Inputs");
 
+            var root = TestUtils.CreateTestSpecificFolder(TestContext);
+            var sourceDirectory = Path.Combine(root, "Source");
+            var destinationDirectory = Path.Combine(root, "Source.tmp");
+            Directory.CreateDirectory(sourceDirectory);
+            File.WriteAllBytes(Path.Combine(sourceDirectory, "ProtoBuf.txt"), new byte[] { });
+
             var projectSnippet = $@"
 <PropertyGroup>
   <SonarQubeTempPath>{rootInputFolder}</SonarQubeTempPath>
+  <RazorCompileOnBuild>true</RazorCompileOnBuild>
+  <ProjectSpecificOutDir>{sourceDirectory}</ProjectSpecificOutDir>
 </PropertyGroup>
 ";
 
-            string afterTargets = string.Join(";", TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+            var afterTargets = string.Join(";", TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
 
             var filePath = CreateProjectFile(null, projectSnippet, afterTargets);
 
             // Act
-            var result = BuildRunner.BuildTargets(TestContext, filePath,
-                TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
 
             // Assert
             result.AssertTargetExecuted(TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
             AssertErrorLogIsSetBySonarQubeTargets(result);
+            File.Exists(Path.Combine(destinationDirectory, "ProtoBuf.txt")).Should().BeTrue();
+        }
+
+        [TestMethod]
```

Comment:
Let's keep those two `ErrorLog` related tests together and move this one below them for logical orientation in the code.